File: solutions/Uncategorized/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int i=0;
        ListNode t=head;
        ListNode k=head;
        while(t!=null)
        {
            i++;
            
            t=t.next;
        }
        
        if(i==1)
        {
            head=null;
            k=head;
            
        }
        
        else if(i==n)
        {
            head=head.next;
            k=head;
        }
        
        else{
        for(int j=1;j<(i-n);j++)
        {
            head=head.next;
        }
           
            head.next=(head.next).next;
            
            
        }
        return k;
    }
}

File: solutions/Uncategorized/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]: 
        curr=head
        z=0
        if head.next==None:
            return None
        while(curr!=None):
            z+=1
            curr=curr.next
        curr=head
        print(curr)
        print(z)
        print(head)
        print(z-n)
        prev=None
        for i in range(z-n-1):
            print(i)
        
            curr=curr.next
        if z-n==0:
            head=head.next
        curr.next=curr.next.next
        return head
